plugins/destination/s3/client/client.go

```diff
@@ -0,0 +1,79 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/cloudquery/cloudquery/plugins/filetypes/csv"
+	"github.com/cloudquery/cloudquery/plugins/filetypes/json"
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	destination.UnimplementedUnmanagedWriter
+	logger     zerolog.Logger
+	spec       specs.Destination
+	pluginSpec Spec
+
+	s3Client   *s3.Client
+	uploader   *manager.Uploader
+	downloader *manager.Downloader
+
+	csvTransformer         *csv.Transformer
+	csvReverseTransformer  *csv.ReverseTransformer
+	jsonTransformer        *json.Transformer
+	jsonReverseTransformer *json.ReverseTransformer
+}
+
+func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (destination.Client, error) {
+	if spec.WriteMode != specs.WriteModeAppend {
+		return nil, fmt.Errorf("destination only supports append mode")
+	}
+	c := &Client{
+		logger:                 logger.With().Str("module", "s3").Logger(),
+		spec:                   spec,
+		csvTransformer:         &csv.Transformer{},
+		jsonTransformer:        &json.Transformer{},
+		csvReverseTransformer:  &csv.ReverseTransformer{},
+		jsonReverseTransformer: &json.ReverseTransformer{},
+	}
+
+	if err := spec.UnmarshalSpec(&c.pluginSpec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal postgresql spec: %w", err)
+	}
+	if err := c.pluginSpec.Validate(); err != nil {
+		return nil, err
+	}
+	c.pluginSpec.SetDefaults()
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load AWS SDK config: %w", err)
+	}
+
+	c.s3Client = s3.NewFromConfig(cfg)
+	c.uploader = manager.NewUploader(c.s3Client)
+	c.downloader = manager.NewDownloader(c.s3Client)
+
+	// we want to run this test because we want it to fail early if the bucket is not accessible
+	if _, err := c.uploader.Upload(ctx, &s3.PutObjectInput{
+		Bucket: aws.String(c.pluginSpec.Bucket),
+		Key:    aws.String("cq-test-file"),
```

Comment:
I wonder if this file might confuse whatever process users use to ingest the data from S3 (Glue Crawler, Databricks, etc). Is it really necessary? The error will surface as soon as the first batch is written anyway. 

If we keep it, we should try and give it some name or property that will make sure it gets ignored by other processes (not sure if that's possible). Or otherwise maybe we should try and delete it? The downside there is that our process may not have `s3:DeleteObject` permissions.

Reply:
Yeah I thought about the deletion but we will have permissions issues for sure. maybe we can try do an empty file ? I don't think there is a good way to test it ?

Reply:
Yeah maybe let's start with an empty file if that works, that should be less confusing to a crawler. Also it might help to have the name start with a `.`, like `.cq-test-file`. Let's start with that and see if it causes any issues